src/Kafka/Consumer.hs

```diff
@@ -62,28 +62,38 @@ newConsumer :: MonadIO m
             => ConsumerProperties
             -> Subscription
             -> m (Either KafkaError KafkaConsumer)
-newConsumer cp (Subscription ts tp) = liftIO $ do
-  kc@(KafkaConf kc' ct) <- newConsumerConf cp
+newConsumer props (Subscription ts tp) = liftIO $ do
+  let cp = setCallback (rebalanceCallback (\_ _ -> return ())) <> props
+  kc@(KafkaConf kc' qref ct) <- newConsumerConf cp
   tp' <- topicConf (TopicProps $ M.toList tp)
   _   <- setDefaultTopicConf kc tp'
-  rdk <- bimap KafkaError Kafka <$> newRdKafkaT RdKafkaConsumer kc'
-  case flip KafkaConsumer kc <$> rdk of
-    Left err -> return $ Left err
-    Right kafka -> do
-      forM_ (cpLogLevel cp) (setConsumerLogLevel kafka)
-      sub <- subscribe kafka ts
-      case sub of
-        Nothing  -> runEventLoop kafka ct (Just $ Timeout 100) >> return (Right kafka)
+  rdk <- newRdKafkaT RdKafkaConsumer kc'
+  case rdk of
+    Left err   -> return . Left $ KafkaError err
+    Right rdk' -> do
+      msgq <- rdKafkaQueueNew rdk'
+      writeIORef qref (Just msgq)
+      let kafka = KafkaConsumer (Kafka rdk') kc
+      redErr <- redirectCallbacksPoll kafka
+      case redErr of
         Just err -> closeConsumer kafka >> return (Left err)
+        Nothing  -> do
+          forM_ (cpLogLevel cp) (setConsumerLogLevel kafka)
+          sub <- subscribe kafka ts
+          case sub of
+            Nothing  -> runConsumerLoop kafka ct (Just $ Timeout 100) >> return (Right kafka)
+            Just err -> closeConsumer kafka >> return (Left err)
 
--- | Polls the next message from a subscription
 pollMessage :: MonadIO m
             => KafkaConsumer
             -> Timeout -- ^ the timeout, in milliseconds
             -> m (Either KafkaError (ConsumerRecord (Maybe BS.ByteString) (Maybe BS.ByteString))) -- ^ Left on error or timeout, right for success
-pollMessage c@(KafkaConsumer (Kafka k) _) (Timeout ms) =
-    liftIO $ pollEvents c Nothing >> rdKafkaConsumerPoll k (fromIntegral ms) >>= fromMessagePtr
-
+pollMessage c@(KafkaConsumer _ (KafkaConf _ qr _)) (Timeout ms) = liftIO $ do
+    pollConsumerEvents c Nothing
+    mbq <- readIORef qr
```

Comment:
Getting the message queue from `KafkaConf` and polling messages from there instead of polling from the _consumer queue_